eth-contracts/contracts/service/interface/ServiceProviderStorageInterface.sol

```diff
@@ -29,20 +30,29 @@ interface ServiceProviderStorageInterface {
   ) external view returns (address owner, string memory endpoint, uint blocknumber, address delegateOwnerWallet);
 
   function getServiceProviderIdFromEndpoint(
-    bytes32 _endpoint
+    string calldata _endpoint
   ) external view returns (uint);
 
-  function getServiceProviderIdFromAddress(
+  function getServiceProviderIdsFromAddress(
     address _ownerAddress,
     bytes32 _serviceType
-  ) external view returns (uint);
+  ) external view returns (uint[] memory);
 
   function getTotalServiceTypeProviders(
     bytes32 _serviceType
   ) external view returns (uint);
-  
+
   function getDelegateOwnerWallet(
     address _owner,
-    bytes32 _serviceType
+    bytes32 _serviceType,
+    string calldata _endpoint
   ) external view returns (address);
+
+  function getStakeAmountFromEndpoint(
```

Comment:
where's the function for this?

Reply:
so this is weirdly not necessary anymore because the total stake amount is just associated with the account and not the endpoint